### PR TITLE
Stage 2 seems to be required after all

### PR DESCRIPTION
**Pull request by camelid**
Reverts most of bb59980b2da10437ce1ee4d53bdb3feb1f4a9c5f.

See the discussion starting at https://github.com/rust-lang/miri/issues/1340#issuecomment-913043714 for more.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -148,11 +148,11 @@ cd rustc
...
-# Build a stage 1 rustc, and build the rustc libraries with that rustc.
+# Build a stage 2 rustc, and build the rustc libraries with that rustc.
 # This step can take 30 minutes or more.
-./x.py build --stage 1 compiler/rustc
+./x.py build --stage 2 compiler/rustc
 # If you change something, you can get a faster rebuild by doing
-./x.py build --keep-stage 0 --stage 1 compiler/rustc
+./x.py build --keep-stage 0 --stage 2 compiler/rustc
 # You may have to change the architecture in the next command
 rustup toolchain link stage1 build/x86_64-unknown-linux-gnu/stage1
```
Comment (RalfJung) [inline CONTRIBUTING.md:157]: This and the next line also need updating.

Comment (camelid) [reply CONTRIBUTING.md:157]: Actually I'm not sure if they do. After I did the `x build --stage 2` build, I tried `./miri build` again, still with my `/stage1` toolchain, and it built successfully. I'll try running `cargo clean` and then building again just to make sure.

Comment (camelid) [reply CONTRIBUTING.md:157]: I think rustbuild has some weird behavior around the interaction between `--stage 2` and `stage1` builds. It's surprising altogether that Miri requires a `--stage 2` build.

Comment (camelid) [reply CONTRIBUTING.md:157]: But I can update it just to be safe if you want, given the previous history of these instructions ;)

Comment (RalfJung) [reply CONTRIBUTING.md:157]: When I completed a stage 1 build, doing the stage 2 build doesn't even do much any more... it just builds libstd (which takes 60s) and then copies a bunch of stuff. Most of stage 2 is a copy of stage 1.

But yeah I'd prefer to consistently use stage 2 here.

Comment (camelid) [reply CONTRIBUTING.md:157]: I `cargo clean`ed and ran `./miri build` again, and it worked, but I'll change it to be consistent.

**Pull request merged**